pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -145,17 +145,34 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gard
 		return err
 	}
 
-	// Reset the `EnableStaticTokenKubeconfig` value to false, when shoot cluster is updated to  k8s version >= 1.27.
-	if versionutils.ConstraintK8sLess127.Check(oldShootKubernetesVersion) && versionutils.ConstraintK8sGreaterEqual127.Check(shootKubernetesVersion) {
-		if ptr.Deref(maintainedShoot.Spec.Kubernetes.EnableStaticTokenKubeconfig, false) {
-			maintainedShoot.Spec.Kubernetes.EnableStaticTokenKubeconfig = ptr.To(false)
+	// Set the .spec.kubernetes.enableStaticTokenKubeconfig field to false, when Shoot cluster is being forcefully updated to K8s >= 1.27.
+	// Gardener forbids enabling the static token kubeconfig for Shoots with K8s 1.27+. See https://github.com/gardener/gardener/pull/7883/commits/ca49644d32df48e0cf3fd00192f2ad078555796c
```

Comment:
Is this not also true for manually updated clusters? In other words: Is `forcefully` being in the comment correct? I would assume that it will always be set to `false` regardless if I trigger the update or it is done by the maintenance controller? Or is the "manual" update handled via some update validation check?

Reply:
> I would assume that it will always be set to false regardless if I trigger the update or it is done by the maintenance controller?

No. If you have 1.26 Shoot with `enableStaticTokenKubeconfig=true` and if you update to 1.27, the validation will fail. See https://github.com/gardener/gardener/blob/0af5ecd480ab84f3768afddaa676c9920f7f4bf3/pkg/apis/core/validation/shoot.go#L838-L840

That's why we need to write such a migration code in the maintenance controller so that it can update Shoots with `enableStaticTokenKubeconfig=true` from 1.26 to 1.27. Otherwise, the patch/update request gets rejected.

> Is this not also true for manually updated clusters? In other words: Is forcefully being in the comment correct?

I used the word `forcefully` to denote a minor K8s version update performed by the maintenance controller.